### PR TITLE
refactor: loadtest

### PR DESCRIPTION
**Pull request by ganchoradkov**
Improved the load test to additionally test message communication between clients

### CODE REVIEW

File: packages/sign-client/package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "build": "run-s build:pre build:cjs build:esm build:umd",
     "test:pre": "rm -rf ./test/test.db",
     "test:run": "env TS_NODE_PROJECT=\"tsconfig.cjs.json\" mocha --timeout 10000 --exit --r ts-node/register ./test/*.spec.ts",
-    "test:concurrency": "env TS_NODE_PROJECT=\"tsconfig.cjs.json\" mocha --timeout 100000 --exit --r ts-node/register ./test/concurrency/*.spec.ts",
+    "test:concurrency": "env TS_NODE_PROJECT=\"tsconfig.cjs.json\" mocha --timeout 1900000 --exit --r ts-node/register ./test/concurrency/*.spec.ts",
```
Comment (arein) [inline packages/sign-client/package.json:29]: nitpick: what is `190000`?

File: packages/sign-client/test/concurrency/concurrency.spec.ts [hunk 2]
```diff
@@ -9,57 +9,102 @@ import {
...
+          const validate = () => {
+            if (messagesReceived.length >= messagesToBeExchanged) {
+              resolve();
+            }
+          };
+
+          for (const i of Array.from(Array(messagesToBeExchanged).keys())) {
+            const client: any = Math.floor(Math.random() * clientsArr.length);
+            await emit(clientsArr[client]);
+            await throttle(10); // throttle the messages/s to avoid being blocked by the relay
```
Comment (arein) [inline packages/sign-client/test/concurrency/concurrency.spec.ts:78]: neat!

File: packages/sign-client/test/concurrency/concurrency.spec.ts [hunk 3]
```diff
@@ -9,57 +9,102 @@ import {
...
 
-          clients.A.emit(eventPayload);
+          const validate = () => {
+            if (messagesReceived.length >= messagesToBeExchanged) {
+              resolve();
+            }
+          };
+
+          for (const i of Array.from(Array(messagesToBeExchanged).keys())) {
+            const client: any = Math.floor(Math.random() * clientsArr.length);
```
Comment (arein) [inline packages/sign-client/test/concurrency/concurrency.spec.ts:76]: i don't get this?

**Review by arein — APPROVED** [review]

File: packages/sign-client/package.json (hunk 1 above)
Comment (ganchoradkov) [reply packages/sign-client/package.json:29]: I've increased the timeout to a higher value to account for increased latency to allow longer tests.
This particular value seemed to work reliably when running large n sockets/messages with local and dev/prod relays

File: packages/sign-client/test/concurrency/concurrency.spec.ts (hunk 3 above)
Comment (ganchoradkov) [reply packages/sign-client/test/concurrency/concurrency.spec.ts:76]: it chooses client randomly to send a message from
so it tests messages between the two clients in both directions
A -> B
B -> A

**Review by xzilja — APPROVED** [review]
Works for now, but we'll need to address these in a away that allows running them without a throttle, i.e. more lightweight approach to client initialisation / reuse in tests

**Pull request merged**